legal-api/src/legal_api/models/filing.py

```diff
@@ -292,6 +293,33 @@ def get_a_businesses_most_recent_filing_of_a_type(business_id: int, filing_type:
 
         return filing.one_or_none()
 
+    @staticmethod
+    def get_most_recent_legal_filing(business_id: str, filing_type: str):
+        """Return the most recent filing containing the legal_filing type."""
+        # Filing._filing_json.has_any(filing_type))).\
+
+        expr = Filing._filing_json[('filing', filing_type)]
+        max_filing = db.session.query(db.func.max(Filing._filing_date).label('last_filing_date')).\
+            filter(Filing.business_id == business_id).\
+            filter(or_(Filing._filing_type == filing_type,
+                       expr.label('legal_filing_type').isnot(None))).\
+            filter(Filing._status == Filing.Status.COMPLETED.value).\
+            subquery()
+
+        filing = Filing.query.join(max_filing, Filing._filing_date == max_filing.c.last_filing_date). \
+            filter(Filing.business_id == business_id). \
+            filter(Filing._status == Filing.Status.COMPLETED.value)
+
+        # As the JSON query is new for most, leaving the debug stmnt
+        # that dumps the query for easier debugging.
+        current_app.logger.debug(
+            str(filing.statement.compile(
+                dialect=dialect(),
+                compile_kwargs={'literal_binds': True}))
+        )
+
+        return filing.one_or_none()
```

Comment:
In my initial test, I get a 500 error from this upon posting. I haven't dug further as to why, but the corpnum in dev is 1523.

Reply:
I think the DB you're using hasn't been migrated. One of the previous commits has the migration that creates the **offices** table. There's also a migration script that updates the data that Peter added.

Reply:
Yeah it's working for me. There's some failing postman tests due to the office table/schema change, but I'll update those now and add them to my PR. My PR includes fixes to the pytests that are failing due to it being 2020 so theres an extra task in the todos

Reply:
Re-opening this - the error I'm getting, over dev db with CP0001523, is sqlalchemy.orm.exc.MultipleResultsFound on the filing.one_or_none. Checked the query from the debug statement in lile 315, and it returns 3 results, hence the error - one COD and two ARs. I haven't tested yet with other corps but this would at least break for this corp, in dev, if it's promoted.